content/automate/ManageIQ/AutomationManagement/AnsibleTower/Operations/StateMachines/Job.class/__methods__/launch_ansible_job.rb

```diff
@@ -116,9 +115,10 @@ def run(job_template, target)
                   args[:limit] = target if target
                   @handle.log(:info, "Job Arguments #{args}")
 
-                  job = @handle.vmdb(JOB_CLASS).create_job(job_template, args)
+                  job = @handle.vmdb(job_template.job_class).create_job(job_template, args)
```

Comment:
Can we avoid having to expose and retain the job class by creating an instance method `create_job` on the service model for `job_template` ?

Since the Job and JobWorkflow objects when you need to load the Job object you can do that from a common class level, like `OrchestrationStack`.

Reply:
Updated based on the comment.